mythx_cli/analyze/util.py

```diff
@@ -1,16 +1,108 @@
 """This module contains helpers for generating MythX analysis payloads."""
 
 import logging
+from enum import Enum
+from glob import glob
 from os.path import abspath, commonpath
 from pathlib import Path
-from typing import Dict
+from typing import Dict, List, Tuple, Union
 
 import click
 
 LOGGER = logging.getLogger("mythx-cli")
 
 
+class AnalyzeMode(Enum):
+    SOLIDITY_FILE = 1
+    SOLIDITY_DIR = 2
+    TRUFFLE = 3
+
+
+def detect_truffle_files(
+    path: Path, project_base: str = "build/contracts/*.json"
+) -> bool:
+    """Detect Truffle projects in paths.
+
+    This function detects whether a Truffle project can be found
+    in the given project base path.
+
+    :param path: The path prefix to look in (e.g. the CLI target)
+    :param project_base: The truffle-specific path suffix
+    :return: Boolean indicating whether path contains a truffle project
+    """
+    return (path / "truffle-config.js").exists() or (path / "truffle.js").exists()
```

Comment:
Should this also check if the project is well formatted? If the contracts are in the correct directory for instance?

Reply:
Checking that all contracts are in the right place would become fairly complex, e.g. when there are node modules and relative import paths involved. I consciously left this out here, because all we need to know is:
- Is the target a truffle project in the first place?
- Does it contain compiled artifacts?

The contract structure can be handled by truffle itself. If the project is valid, it will compile and we will see artifacts. If it's not, but a truffle config file exists, we throw an error later on. :+1: